maths/prime_number.rb

```diff
@@ -0,0 +1,31 @@
+# A ruby program to check a given number is prime or not
+# Mathematical explanation: A number which has only 2 factors i.e., 1 (one) and itself
+
+# Prime number check function
+def is_prime_number(number)
+  prime_flag = false
+  if number>0
+    if number == 1
+      prime_flag = true
+    else
+      prime_flag = (2..number/2).any? { |i| number % i == 0 }
```

Comment:
**[Optimization]**

We can skip dividing by even factors if the number is not even, like this.

```rb
if number % 2 == 0
  prime_flag = true
else
  (3..number/2).step(2).any? { |i| number % i == 0 }
end
```

-  `(3..number/2).step(2)` yields 3, 5, 7, 9...

Reply:
`if number % 2 == 0
    prime_flag = true`
The above conditional statement may create some conflict with input 2. I have tested it, and it failed to show input number 2 as a prime number.

Reply:
Sorry. I made a mistake!

This one will work.

```rb
if number == 2 # 2 is even but a prime number.
  prime_flag = false
elsif number % 2 == 0
  prime_flag = true
else
  # Assign a return value from any? to prime_flag
  prime_flag = (3..number/2).step(2).any? { |i| number % i == 0 }
end
```